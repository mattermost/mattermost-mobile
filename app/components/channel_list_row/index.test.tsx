// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithEverything} from '@test/intl-test-helper';
import TestHelper from '@test/test_helper';

import ChannelListRow from '.';

import type Database from '@nozbe/watermelondb/Database';

describe('components/channel_list_row', () => {
    let database: Database;
    const channel: Channel = {
        id: '1',
        create_at: 1111,
        update_at: 1111,
        delete_at: 0,
        team_id: 'my team',
        type: 'O',
        display_name: 'channel',
        name: 'channel',
        header: 'channel',
        purpose: '',
        last_post_at: 1,
        total_msg_count: 1,
        extra_update_at: 1,
        creator_id: '1',
        scheme_id: null,
        group_constrained: null,
        shared: true,
    };

    beforeAll(async () => {
        const server = await TestHelper.setupServerDatabase();
        database = server.database;
    });

    it('should match snapshot with open channel icon', () => {
        const wrapper = renderWithEverything(
            <ChannelListRow
                channel={channel}
                selected={false}
                selectable={false}
                testID='ChannelListRow'
                onPress={() => {
                    // noop
                }}
            />,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with private channel icon', () => {
        channel.type = 'P';

        const wrapper = renderWithEverything(
            <ChannelListRow
                channel={channel}
                selected={false}
                selectable={false}
                testID='ChannelListRow'
                onPress={() => {
                    // noop
                }}
            />,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with delete_at filled in', () => {
        channel.delete_at = 1111;
        channel.shared = false;
        channel.type = 'O';

        const wrapper = renderWithEverything(
            <ChannelListRow
                channel={channel}
                testID='ChannelListRow'
                selectable={false}
                selected={false}
                onPress={() => {
                    // noop
                }}
            />,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with shared filled in', () => {
        channel.delete_at = 0;
        channel.shared = true;
        channel.type = 'O';

        const wrapper = renderWithEverything(
            <ChannelListRow
                channel={channel}
                testID='ChannelListRow'
                selectable={false}
                selected={false}
                onPress={() => {
                    // noop
                }}
            />,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with purpose filled in', () => {
        channel.purpose = 'My purpose';

        const wrapper = renderWithEverything(
            <ChannelListRow
                channel={channel}
                testID='ChannelListRow'
                selectable={false}
                selected={false}
                onPress={() => {
                    // noop
                }}
            />,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should be selected', () => {
        const wrapper = renderWithEverything(
            <ChannelListRow
                channel={channel}
                testID='ChannelListRow'
                selectable={true}
                selected={true}
                onPress={() => {
                    // noop
                }}
            />,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
