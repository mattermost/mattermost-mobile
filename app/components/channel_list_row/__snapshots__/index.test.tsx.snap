// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/channel_list_row should be selected 1`] = `
<View
  style={
    {
      "paddingVertical": 9,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="ChannelListRow.channel"
    >
      <Icon
        name="circle-multiple-outline"
        size={20}
        style={
          {
            "color": "rgba(63,67,80,0.56)",
            "padding": 2,
          }
        }
      />
      <View
        style={
          {
            "flexDirection": "column",
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#3f4350",
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
          testID="ChannelListRow.channel.display_name"
        >
          channel
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": "rgba(63,67,80,0.64)",
              "fontFamily": "OpenSans",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
            }
          }
        >
          My purpose
        </Text>
      </View>
      <View>
        <Icon
          color="#1c58d9"
          name="check-circle"
          size={28}
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`components/channel_list_row should match snapshot with delete_at filled in 1`] = `
<View
  style={
    {
      "paddingVertical": 9,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="ChannelListRow.channel"
    >
      <Icon
        name="archive-outline"
        size={20}
        style={
          {
            "color": "rgba(63,67,80,0.56)",
            "padding": 2,
          }
        }
      />
      <View
        style={
          {
            "flexDirection": "column",
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#3f4350",
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
          testID="ChannelListRow.channel.display_name"
        >
          channel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/channel_list_row should match snapshot with open channel icon 1`] = `
<View
  style={
    {
      "paddingVertical": 9,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="ChannelListRow.channel"
    >
      <Icon
        name="circle-multiple-outline"
        size={20}
        style={
          {
            "color": "rgba(63,67,80,0.56)",
            "padding": 2,
          }
        }
      />
      <View
        style={
          {
            "flexDirection": "column",
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#3f4350",
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
          testID="ChannelListRow.channel.display_name"
        >
          channel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/channel_list_row should match snapshot with private channel icon 1`] = `
<View
  style={
    {
      "paddingVertical": 9,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="ChannelListRow.channel"
    >
      <Icon
        name="circle-multiple-outline"
        size={20}
        style={
          {
            "color": "rgba(63,67,80,0.56)",
            "padding": 2,
          }
        }
      />
      <View
        style={
          {
            "flexDirection": "column",
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#3f4350",
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
          testID="ChannelListRow.channel.display_name"
        >
          channel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/channel_list_row should match snapshot with purpose filled in 1`] = `
<View
  style={
    {
      "paddingVertical": 9,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="ChannelListRow.channel"
    >
      <Icon
        name="circle-multiple-outline"
        size={20}
        style={
          {
            "color": "rgba(63,67,80,0.56)",
            "padding": 2,
          }
        }
      />
      <View
        style={
          {
            "flexDirection": "column",
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#3f4350",
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
          testID="ChannelListRow.channel.display_name"
        >
          channel
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": "rgba(63,67,80,0.64)",
              "fontFamily": "OpenSans",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
            }
          }
        >
          My purpose
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/channel_list_row should match snapshot with shared filled in 1`] = `
<View
  style={
    {
      "paddingVertical": 9,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="ChannelListRow.channel"
    >
      <Icon
        name="circle-multiple-outline"
        size={20}
        style={
          {
            "color": "rgba(63,67,80,0.56)",
            "padding": 2,
          }
        }
      />
      <View
        style={
          {
            "flexDirection": "column",
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#3f4350",
              "fontFamily": "OpenSans",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
          testID="ChannelListRow.channel.display_name"
        >
          channel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
